dotcom-rendering/src/web/lib/useDiscussion.ts

```diff
@@ -51,7 +51,11 @@ type CommentType = {
 };
 
 export const useDiscussion = (url: string) => {
-	const { data } = useApi<DiscussionResponse>(url);
+	const { data } = useApi<DiscussionResponse>(url, {
+		// The default for dedupingInterval is 2 seconds but we want to wait longer here because the cache time
+		// for a discussion is at least 15 seconds
+		dedupingInterval: 8000,
+	});
```

Comment:
This is the magic. The reason this PR is possible is because the SWR library dedupes multiple api calls to the same endpoint. So, yes, now that we're breaking `CommentCount` out into its own Island and now that it is making its own api call to get the discussion, this means **we are making the same api call twice**. But this is fine because there is actually only one _network_ request made.

### Why 8 seconds?
The default is 2 but because we're already caching the discussion endpoint calls we may as well extend the time we dedupe them to reduce network round trips.

Reply:
Just to check, do we know that the SWR library dedupes/caches calls from different react renders/instances? I'm guessing this was checked in the dev tools? :)

Reply:
This is a very good question. I spent some time checking this and found that:

1. It will _not_ dedupe between an `Island` and `App.tsx`
2. But yes, it will dedupe over multiple `Islands`

SWR uses an in memory cache which gets reinitialised inside `BootReact` but the separate island instances share one. I did look into if there was a way to create one shared global cache but it was not possible

Reply:
Potential bikeshed: Is there a test to check this?

Reply:
I do want to test this but the only way we can really achieve that is with a Cypress test and for that I need a better use case. I've got a future PR coming up that will add the second api for this same endpoint to an Island as well so that would be a good moment to add this test

Reply:
Ah, great. I was thinking of a test that ensured we have SWR worked as expected with islands, but that could be a little close to testing the SWR functionality, but would give us confidence in the question above. It would also then define that as our line in the sand and be able to swap out SWR with confidence at a later date.

Reply:
My aim is to load a page that has comments on and then count the number of api requests that get made to this endpoint. The test will expect there to only be one. This test will pass now because we centralise the call and share state and the same test will pass with SWR because that lib solves the same problem another way (using a cache).

Ideally here we would add the test in _before_ this PR so we could show deduplication working before and after but because of the point I made earlier where SWR can't share a cache between App and the Islands, that test would actually fail with this PR and only start working again with the next one